### PR TITLE
feat: #220 updated

### PR DESCRIPTION
**Pull request by Zekfad**
A little rework made of #220.
Fixed deprecated API usage.
Successfully compilers on Windows 10 (x64) Enterprise.

### CODE REVIEW

**Comment by TheRealDannyyy** [comment]
**Thanks a lot for doing this @Zekfad!**

I think the only thing left that will be requested before the merge, are docs for the added stuff.
Would you mind adding the docs? (Tagging @ixevix or @patrickklug for possible review and merge.)

**Comment by Zekfad** [comment]
I guess, I'll work on it within a week, just have some work to do right now.

**Comment by Zekfad** [comment]
@TheRealDannyyy, docs are here now.

@ixevix, @patrickklug, please review.

**Comment by Zekfad** [comment]
Oh and this PR should also close #222.

**Comment by ixevix** [comment]
Looks good to me

**Comment by Zekfad** [comment]
Oh, and cuz most of that stuff is rewrite of #220, any ideas how to mention original PR author contributions?

**Comment by TheRealDannyyy** [comment]
Thanks a lot for your work @Zekfad! 

Can't wait to make use of this and share it with other game developers! I'm not familiar with the code. Does this work with the Steam SDK v1.42 or do we need to use a more recent version now?

**Comment by Zekfad** [comment]
@TheRealDannyyy, I guess it gonna work if no new features was released.
Rich Presence was released long ago, in v1.13 (26th April 2011), what about another features - I don't really know, but I'd still suggest to just update SDK if possible in order to mitigate possible errors.
Also, if you want I can try to compile it with that version in case I'll find binaries or you'll provide them.

**Comment by TheRealDannyyy** [comment]
We already set up an automated builder and [downloader website](https://greenworks-prebuilds.armaldio.xyz/). I guess once this is merged, we'll see if things work as expected or if an SDK update is necessary.

**Comment by patrickklug** [comment]
@hokein can you merge if you are happy with this?

File: src/api/steam_api_settings.cc [hunk 1]
```diff
@@ -334,6 +340,7 @@ void RegisterAPIs(v8::Local<v8::Object> target) {
   SET_FUNCTION("getImageSize", GetImageSize);
   SET_FUNCTION("getImageRGBA", GetImageRGBA);
   SET_FUNCTION("getIPCountry", GetIPCountry);
+  SET_FUNCTION("runCallbacks", RunCallbacks);
```
Comment (hokein) [inline src/api/steam_api_settings.cc:343]: I think we should not expose this function, it is internal.

**Comment by hokein** [comment]
@Zekfad nice work! mostly good, but it seems that we miss the documents for the APIs in steam_api_matchmaking.cc, could you add that?

btw, I believe you have built it successfully. Have you used these APIs in your App/game?

File: src/api/steam_api_settings.cc (hunk 1 above)
Comment (Zekfad) [reply src/api/steam_api_settings.cc:343]: Steam API reference suggesting to call it every render frame manually:

> For callbacks to dispatch to registered listeners you must call the SteamAPI_RunCallbacks. It's best to call this frequently, as the more time between calls, the more latency between receiving events or results from the Steamworks API. **Most games call this once per render-frame, it's highly recommended that you call this at least once a second.** All registered listener functions will be invoked during this call, in the thread context where SteamAPI_RunCallbacks was called from.

[Source](https://partner.steamgames.com/doc/sdk/api#callbacks)

And in source there's no any loops calling this functions, any ideas?

Comment (hokein) [reply src/api/steam_api_settings.cc:343]: yeah, but in greenworks, it is called automatically, so there is no need for users to call it, https://github.com/greenheartgames/greenworks/blob/e18f7bc08e42c2716e2be55bc1c378057d398752/src/steam_client.cc#L148.

Comment (Zekfad) [reply src/api/steam_api_settings.cc]: Ah, sorry, didn't saw that. I guess it can be resolved now with 83e993a commit.

**Comment by Zekfad** [comment]
@hokein, ah, sorry I totally forgot about `steam_api_matchmaking.cc`, I'll add it within a day or two.
Yeah I used it on my side, but I was needed for Rich Presence, so that's the only thing I really tested so far.

**Comment by Zekfad** [comment]
I had too much work to do. Now I'm on my way to make docs.
Do I need to specify that matchmaking is untested and experimental part of library in docs?

**Comment by Zekfad** [comment]
@hokein, docs are here, please made a review.

**Comment by hokein** [comment]
Thanks, looks great!

**Pull request merged**